src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -1162,118 +1164,248 @@ class ObservationStore(
 
-      with(OBSERVED_ZONE_SPECIES_TOTALS) {
-        dslContext
-            .deleteFrom(OBSERVED_ZONE_SPECIES_TOTALS)
-            .where(OBSERVATION_ID.eq(observationId))
-            .and(SPECIES_NAME.eq(otherSpeciesName))
-            .and(CERTAINTY_ID.eq(RecordedSpeciesCertainty.Other))
-            .execute()
-      }
+      dslContext
+          .update(OBSERVATION_BIOMASS_QUADRAT_SPECIES)
+          .set(BIOMASS_SPECIES_ID, targetBiomassSpeciesId)
+          .where(BIOMASS_SPECIES_ID.eq(otherBiomassSpeciesId))
+          .andNotExists(
+              DSL.selectOne()
+                  .from(quadratSpeciesTable2)
+                  .where(quadratSpeciesTable2.BIOMASS_SPECIES_ID.eq(targetBiomassSpeciesId))
+                  .and(quadratSpeciesTable2.POSITION_ID.eq(POSITION_ID)))
+          .execute()
 
-      with(OBSERVED_SITE_SPECIES_TOTALS) {
+      dslContext
+          .update(OBSERVATION_BIOMASS_QUADRAT_SPECIES)
+          .set(
+              ABUNDANCE_PERCENT,
+              ABUNDANCE_PERCENT.plus(
+                  DSL.field(
+                      DSL.select(quadratSpeciesTable2.ABUNDANCE_PERCENT)
+                          .from(quadratSpeciesTable2)
+                          .where(quadratSpeciesTable2.BIOMASS_SPECIES_ID.eq(otherBiomassSpeciesId))
+                          .and(quadratSpeciesTable2.POSITION_ID.eq(POSITION_ID)))))
+          .where(BIOMASS_SPECIES_ID.eq(targetBiomassSpeciesId))
+          .andExists(
+              DSL.selectOne()
+                  .from(quadratSpeciesTable2)
+                  .where(quadratSpeciesTable2.BIOMASS_SPECIES_ID.eq(otherBiomassSpeciesId))
+                  .and(quadratSpeciesTable2.POSITION_ID.eq(POSITION_ID)))
+          .execute()
+    }
+
+    // Invasive and threatened should be true if they're true on either version of the species.
+    with(OBSERVATION_BIOMASS_SPECIES) {
+      dslContext
+          .update(OBSERVATION_BIOMASS_SPECIES)
+          .set(
+              IS_INVASIVE,
+              DSL.select(DSL.boolOr(IS_INVASIVE))
+                  .from(OBSERVATION_BIOMASS_SPECIES)
+                  .where(ID.`in`(otherBiomassSpeciesId, targetBiomassSpeciesId)))
+          .set(
+              IS_THREATENED,
+              DSL.select(DSL.boolOr(IS_THREATENED))
+                  .from(OBSERVATION_BIOMASS_SPECIES)
+                  .where(ID.`in`(otherBiomassSpeciesId, targetBiomassSpeciesId)))
+          .where(ID.eq(targetBiomassSpeciesId))
+          .execute()
+    }
+
+    // ON DELETE CASCADE will remove the data for the "Other" species from all the biomass tables.
+    dslContext
+        .deleteFrom(OBSERVATION_BIOMASS_SPECIES)
+        .where(OBSERVATION_BIOMASS_SPECIES.ID.eq(otherBiomassSpeciesId))
+        .execute()
+  }
+
+  private fun mergeOtherSpeciesForMonitoring(
+      observation: ExistingObservationModel,
+      speciesId: SpeciesId,
+      otherSpeciesName: String
+  ) {
+    val observationId = observation.id
+    val observationPlotDetails =
```

Comment:
No changes to the code here, just moving it from `mergeOtherSpecies` to this private method.